contracts/marketplace/src/execute.rs

```diff
@@ -691,6 +703,40 @@ pub fn execute_accept_bid(
     Ok(res.add_event(event))
 }
 
+pub fn execute_reject_bid(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    collection: Addr,
+    token_id: TokenId,
+    bidder: Addr,
+) -> Result<Response, ContractError> {
+    nonpayable(&info)?;
+    only_owner(deps.as_ref(), &info, &collection, token_id)?;
+
+    let bid_key = bid_key(&collection, token_id, &bidder);
+
+    let bid = bids().load(deps.storage, bid_key.clone())?;
+    if bid.is_expired(&env.block) {
+        return Err(ContractError::BidExpired {});
+    }
+
+    bids().remove(deps.storage, bid_key)?;
+
+    let refund_msg = BankMsg::Send {
+        to_address: bidder.to_string(),
+        amount: vec![coin(bid.price.u128(), NATIVE_DENOM.to_string())],
+    };
+
+    let event = Event::new("reject-bid")
+        .add_attribute("collection", collection.to_string())
+        .add_attribute("token_id", token_id.to_string())
+        .add_attribute("bidder", bidder)
+        .add_attribute("price", bid.price.to_string());
+
+    Ok(Response::new().add_event(event).add_message(refund_msg))
```

Comment:
This should be triggering the hook as in remove_bid
https://github.com/public-awesome/marketplace/blob/19f337920f4a186550e2441ae88221030d14d844/contracts/marketplace/src/execute.rs#L597-L629